vendor/ddcloud/resource_additional_nics_test.go

```diff
@@ -232,7 +232,13 @@ func testCheckDDCloudAdditionalNicMatchesIPV4(serverResourceName string, expecte
 			return fmt.Errorf("Bad: Server '%s' has no additional nics", serverID)
 		}
 
-		if *server.Network.AdditionalNetworkAdapters[0].PrivateIPv4Address != expected {
+		isNicExists := false
+		for _, networkAdapters := range server.Network.AdditionalNetworkAdapters {
+			if *networkAdapters.PrivateIPv4Address == expected {
+				isNicExists = true
+			}
+		}
+		if !isNicExists {
 			return fmt.Errorf("Bad: Server '%s' has additional nic with the ip address %s (expected %s) ", serverID, server.Network.AdditionalNetworkAdapters[0].PrivateIPv4Address, expected)
```

Comment:
Does this message still makes sense given the change in logic? Since there can be multiple additional NICs, perhaps it might be more correct to note that the desired NIC was not found, rather than listing the NIC that _was_ found.